tests/unitary/src/compute/allocator/test_cuda_memory_block_pool.cpp

```diff
@@ -0,0 +1,521 @@
+generate_partitions(cuda_memory_block_pool &pool, std::size_t number)
+{
+    cuda_memory_block_pool::iterator ite;
+    cuda_memory_block_pool::iterator result;
+    const cuda_memory_block_pool::iterator null;
+    const std::size_t size = 1024;
+    std::uintptr_t address = 0xDEADBEEF;
+
+    for (std::size_t i = 0; i < number; ++i)
+    {
+        std::tie(ite, std::ignore) = pool.emplace(
+            std::piecewise_construct,
+            std::forward_as_tuple(reinterpret_cast<void*>(address), size, 16),
+            std::forward_as_tuple(ite, null, false)
+        );
+        update_backward_link(ite);
+        address += size;
+
+        if (i == 0)
+        {
+            result = ite;
+        }
+    }
+
+    return result;
+}
+
+static void invalidate_forward_link(cuda_memory_block_pool::iterator ite)
+{
+    const cuda_memory_block_pool::iterator null;
+    ite->second.get_next_block()->second.set_previous_block(null);
+}
+
+static void invalidate_backward_link(cuda_memory_block_pool::iterator ite)
+{
+    const cuda_memory_block_pool::iterator null;
+    ite->second.get_previous_block()->second.set_next_block(null);
+}
+
+
+
+TEST_CASE( "is_partition should return false when not partitioned", "[cuda_memory_block_pool]" )
+{
+    cuda_memory_block_pool pool;
+    auto ite = generate_partitions(pool, 1);
+
+    REQUIRE( is_partition(ite->second) == false );
+}
+
+TEST_CASE( "is_partition should return true when partitioned in two", "[cuda_memory_block_pool]" )
+{
+    cuda_memory_block_pool pool;
+    auto ite = generate_partitions(pool, 2);
+
+    // Check for both halves that the condition is satisfied
+    REQUIRE( is_partition(ite->second) == true );
+    ite = ite->second.get_next_block();
+    REQUIRE( is_partition(ite->second) == true );
+}
+
+TEST_CASE( "is_partition should return true when partitioned in three", "[cuda_memory_block_pool]" )
+{
+    cuda_memory_block_pool pool;
+    auto ite = generate_partitions(pool, 3);
+
+    // Check for all thirds that the condition is satisfied
+    REQUIRE( is_partition(ite->second) == true );
+    ite = ite->second.get_next_block();
+    REQUIRE( is_partition(ite->second) == true );
+    ite = ite->second.get_next_block();
+    REQUIRE( is_partition(ite->second) == true );
+}
+
+TEST_CASE( "is_mergeable should return false when null iterator", "[cuda_memory_block_pool]" )
+{
+    const cuda_memory_block_pool::iterator null;
+    REQUIRE( is_mergeable(null) == false );
+}
+
+TEST_CASE( "is_mergeable should return false when occupied", "[cuda_memory_block_pool]" )
+{
+    cuda_memory_block_pool pool;
+    auto ite = generate_partitions(pool, 1);
+    ite->second.set_free(false);
+    REQUIRE( is_mergeable(ite) == false );
+}
+
+TEST_CASE( "is_mergeable should return true when free", "[cuda_memory_block_pool]" )
+{
+    cuda_memory_block_pool pool;
+    auto ite = generate_partitions(pool, 1);
+    ite->second.set_free(true);
+    REQUIRE( is_mergeable(ite) == true );
+}
```

Comment:
test case when more than 1 partition

Reply:
This function only analyzes the item passed to it, not adjacent ones. Thus, partitions do not affect its behavior.